Mage.Sets/src/mage/cards/m/MirriWeatherlightDuelist.java

```diff
@@ -83,7 +83,8 @@ public boolean canBlock(Permanent attacker, Permanent blocker, Ability source, G
             return true;
         }
         for (UUID creature : game.getCombat().getBlockers()) {
-            if (game.getPlayer(game.getPermanent(creature).getOwnerId()).hasOpponent(attacker.getControllerId(), game)) {
+            if (game.getPlayer(game.getPermanent(creature).getControllerId()).hasOpponent(attacker.getControllerId(), game) &&
+                game.getPermanent(creature).isControlledBy(blocker.getControllerId())) {
```

Comment:
I keep reading this over and over, but I still don't have a good mental picture of what each of these do. I think these needs a rewrite to be understandable. I think it'd start with `creature` -> `creatureId` and by declaring a local variable to hold `game.getPermanent(creature)`

Reply:
Done. Renamed 'blocker' to 'newBlocker', added 'existingBlocker' for the creature we're inspecting inside the loop.